### ISSUE TITLE
haotian-liu/LLaVA: [NeurIPS'23 Oral] Visual Instruction Tuning (LLaVA) built towards GPT-4V level capabilities and beyond.

### ISSUE THREAD

**Issue by irthomasthomas** [irthomasthomas/undecidability#597]
- [ ] [haotian-liu/LLaVA: [NeurIPS'23 Oral] Visual Instruction Tuning (LLaVA) built towards GPT-4V level capabilities and beyond.](https://github.com/haotian-liu/LLaVA?tab=readme-ov-file#demo)

# haotian-liu/LLaVA: [NeurIPS'23 Oral] Visual Instruction Tuning (LLaVA)

**DESCRIPTION:**  
- [1/30] 🔥 LLaVA-NeXT (LLaVA-1.6) is out! With additional scaling to LLaVA-1.5, LLaVA-NeXT-34B outperforms Gemini Pro on some benchmarks. It can now process 4x more pixels and perform more tasks/applications than before. Check out the blog post, and explore the demo! Models are available in Model Zoo. Training/eval data and scripts coming soon.
- [11/10] LLaVA-Plus is released: Learning to Use Tools for Creating Multimodal Agents, with LLaVA-Plus (LLaVA that Plug and Learn to Use Skills). [Project Page] [Demo] [Code] [Paper]
- [11/2] LLaVA-Interactive is released: Experience the future of human-AI multimodal interaction with an all-in-one demo for Image Chat, Segmentation, Generation and Editing. [Project Page] [Demo] [Code] [Paper]
- [10/26] 🔥 LLaVA-1.5 with LoRA achieves comparable performance as full-model finetuning, with a reduced GPU RAM requirement (ckpts, script). We also provide a doc on how to finetune LLaVA-1.5 on your own dataset with LoRA.

**URL:**  
[haotian-liu/LLaVA](https://github.com/haotian-liu/LLaVA?tab=readme-ov-file#demo)

#### {'label-name': 'Multimodal Learning', 'label-description': 'Content related to machine learning models that process and generate multiple types of data simultaneously.', 'confidence': 66.24}

**Comment by irthomasthomas** [comment]
### Related issues

### #184: Robin: Multimodal (Visual-Language) Models.  - CERC-AAI Lab - Robin v1.0
<details><summary>### Details</summary>Similarity score: 0.9
- [ ] [CERC-AAI Lab - Robin v1.0](https://sites.google.com/view/irinalab/blog/robin-v1-0)

    The Robin team is proud to present Robin, a suite of  Multimodal (Visual-Language) Models. 
These models outperform, or perform on par with, the state of the art models of similar scale. 
In the ever-evolving realm of artificial intelligence, the intersection of language understanding and visual perception has paved the way for groundbreaking multimodal models. We study different components and methods to merge pretrained vision and language models with the goal to build better visual language models. 
As part of this first milestone, we release this LLaVA-fork enabling the Mistral-7B & Open-Hermes-2.5 language models to process images. We combine the pretrained LLMs (Vicuna, Mistral and OpenHermes 2.5) and Vision models (CLIP and SigLIP), and further enhance capabilities by finetuning the vision encoder.

Models detailed bellow are available here: https://huggingface.co/agi-collective
The code used is available here: https://github.com/AGI-Collective/Robin/releases/tag/v1.0.0
Also, some related work by our team on aligning multimodal models: https://arxiv.org/abs/2304.13765
LLaVA Architecture Overview
The LLaVA architecture, an acronym for Large Language and Vision Assistant, represents a multimodal Visual Language Model (VLM). At its core, LLaVA integrates a pretrained language model with a pretrained vision encoder, connected through a projection layer. In its original incarnation, the Vicuna model served as the language foundation, while the CLIP ViT-Large from OpenAI assumes the role of the vision encoder.
Building upon this foundation, as part of the first milestone we study the impact of different language models, vision encoders and the effect of finetuning the vision encoder on the performance of our multimodal model. Notably, our journey led us to experiment with the fusion of various versions of the Mistral AI LLM model and the DeepMind SigLip visual encoder.
Architecture Variations
Our model variations are best encapsulated in the table below, outlining the diverse combinations of language models, vision encoders and the fine-tuning strategy.</details>


### #459: llama2
<details><summary>### Details</summary>Similarity score: 0.88
- [ ] [llama2](https://ollama.ai/library/llama2)

Llama 2
========

The most popular model for general use.

*265.8K Pulls*
*Updated 4 weeks ago*

Overview
--------

Llama 2 is released by Meta Platforms, Inc. This model is trained on 2 trillion tokens, and by default supports a context length of 4096. Llama 2 Chat models are fine-tuned on over 1 million human annotations, and are made for chat.

CLI
---

Open the terminal and run

```bash
ollama run llama2
```

API
---

Example using curl:

```bash
curl -X POST http://localhost:11434/api/generate -d '{
  "model": "llama2",
  "prompt":"Why is the sky blue?"
 }'
```

API documentation
-----------------

Memory requirements
-------------------

-  7b models generally require at least 8GB of RAM
-  13b models generally require at least 16GB of RAM
-  70b models generally require at least 64GB of RAM

If you run into issues with higher quantization levels, try using the q4 model or shut down any other programs that are using a lot of memory.

Model variants
--------------

-   **Chat**: fine-tuned for chat/dialogue use cases. These are the default in Ollama, and for models tagged with `-chat` in the tags tab.

    Example: `ollama run llama2`

-   **Pre-trained**: without the chat fine-tuning. This is tagged as `-text` in the tags tab.

    Example: `ollama run llama2:text`

By default, Ollama uses 4-bit quantization. To try other quantization levels, please use the other tags. The number after the `q` represents the number of bits used for quantization (i.e. `q4` means 4-bit quantization). The higher the number, the more accurate the model is, but the slower it runs, and the more memory it requires.

References
----------

-   [Llama 2: Open Foundation and Fine-Tuned Chat Models](https://metastring.com/llama2)
-   [Meta’s Hugging Face repo](https://huggingface.co/Meta)

#### Suggested labels
#### {   "label-name": "llama2-model",   "description": "A powerful text model for chat, dialogue, and general use.",   "repo": "ollama.ai/library/llama2",   "confidence": 91.74 }</details>


### #152: Summary of Hacker News Forum Post: "Bash one-liners for LLMs"
<details><summary>### Details</summary>Similarity score: 0.88
# Summary of Hacker News Discussion: Bash One-Liners for LLMs

## Overview
This document provides a summary of key comments from a Hacker News forum post titled "Bash one-liners for LLMs." The discussion centers around the use of Large Language Models (LLMs) in conjunction with Command Line Interfaces (CLIs).

## Key Comments and Links

### Simonw's Tools
- **LLM Datasette**: [llm.datasette.io](https://llm.datasette.io) - A CLI utility and Python library for interacting with LLMs.
- **BLIP-Caption**: [blip-caption on GitHub](https://github.com/simonw/blip-caption) - Generate captions for images with Salesforce BLIP.
- **OSpeak**: [ospeak on GitHub](https://github.com/simonw/ospeak) - CLI tool for running text through OpenAI Text to speech.

### Larve's Tools
- **Geppetto**: [geppetto on GitHub](https://github.com/go-go-golems/geppetto) - Golang GPT3 tooling with CLI and TUI chat mode.
- **Prompto**: [prompto on GitHub](https://github.com/go-go-golems/prompto) - Quickly get custom prompt contexts.
- **Promptos**: [promptos on GitHub](https://github.com/go-go-golems/promptos) - A collection of promptos for third-party packages.

### Jart's Contributions
- **Llama.cpp Example**: [Llama.cpp Example on GitHub](https://github.com/ggerganov/llama.cpp/blob/master/examples/Miku.sh) - Example script from the llama.cpp GitHub repository.
- **Shell Script for Renaming Images**: [Shell Script on Gist](https://gist.github.com/jart/bd2f603aefe6ac8004e6b709223881c0) - A script using LLMs for renaming images.

### Verdverm's Tool
- **ChatGPT CLI**: [chatgpt on GitHub](https://github.com/verdverm/chatgpt) - CLI application for working with ChatGPT.

### Alchemist1e9's Tools
- **Chatblade**: [chatblade on GitHub](https://github.com/npiv/chatblade) - CLI Swiss Army Knife for ChatGPT.
- **SGPT**: [sgpt on GitHub](https://github.com/tbckr/sgpt) - Command-line tool for interacting with OpenAI models.

### Additional References
- **Embeddings Blog Post**: [Simon Willison's Blog](https://simonwillison.net/2023/Oct/23/embeddings/) - Discussing embeddings in LLMs.
- **FreeLing User Manual**: [Splitter Module](https://freeling-user-manual.readthedocs.io) - Information about the Splitter Module in FreeLing.
- **Unstructured-IO**: [unstructured on GitHub](https://github.com/Unstructured-IO/unstructured) - Libraries and APIs for custom ML pipelines.
- **Actually Portable Executable**: [Justine's Blog](https://justine.lol/ape.html) - Discussion on systems engineering and microprocessor architecture.
- **NVIDIA GPU Accelerated Computing on WSL 2**: [NVIDIA Documentation](https://docs.nvidia.com/cuda/wsl-user-guide/index.html) - Setting up and troubleshooting CUDA on WSL 2.

## Conclusion
The discussion reflects a vibrant ecosystem surrounding the use of LLMs and CLIs, focusing on ease of use, extensibility, and integration into various workflows.

For more detailed information, visit each link provided above.
</details>


### #305: Home - LibreChat
<details><summary>### Details</summary>Similarity score: 0.87
- [ ] [Home - LibreChat](https://docs.librechat.ai/index.html)


Table of contents
🪶 Features
📃 All-In-One AI Conversations with LibreChat
⭐ Star History
✨ Contributors
💖 This project exists in its current state thanks to all the people who contribute
 

LibreChat

       

🪶 Features

🖥️ UI matching ChatGPT, including Dark mode, Streaming, and 11-2023 updates
💬 Multimodal Chat:
Upload and analyze images with GPT-4 and Gemini Vision 📸
More filetypes and Assistants API integration in Active Development 🚧
🌎 Multilingual UI:
English, 中文, Deutsch, Español, Français, Italiano, Polski, Português Brasileiro, Русский
日本語, Svenska, 한국어, Tiếng Việt, 繁體中文, العربية, Türkçe, Nederlands
🤖 AI model selection: OpenAI API, Azure, BingAI, ChatGPT, Google Vertex AI, Anthropic (Claude), Plugins
💾 Create, Save, & Share Custom Presets
🔄 Edit, Resubmit, and Continue messages with conversation branching
📤 Export conversations as screenshots, markdown, text, json.
🔍 Search all messages/conversations
🔌 Plugins, including web access, image generation with DALL-E-3 and more
👥 Multi-User, Secure Authentication with Moderation and Token spend tools
⚙️ Configure Proxy, Reverse Proxy, Docker, many Deployment options, and completely Open-Source
📃 All-In-One AI Conversations with LibreChat

LibreChat brings together the future of assistant AIs with the revolutionary technology of OpenAI's ChatGPT. Celebrating the original styling, LibreChat gives you the ability to integrate multiple AI models. It also integrates and enhances original client features such as conversation and message search, prompt templates and plugins.

With LibreChat, you no longer need to opt for ChatGPT Plus and can instead use free or pay-per-call APIs. We welcome contributions, cloning, and forking to enhance the capabilities of this advanced chatbot platform.

#### Suggested labels
#### "ai-platform"</details>
